crates/polars-core/src/frame/asof_join/groups.rs

```diff
@@ -633,7 +633,12 @@ fn dispatch_join<T: PolarsNumericType>(
     tolerance: Option<AnyValue<'static>>,
 ) -> PolarsResult<Vec<Option<IdxSize>>> {
     let out = if left_by.width() == 1 {
-        match left_by_s.dtype() {
+        let left_dtype = left_by_s.dtype();
```

Comment:
Looks like a dtype check was missed when the `by` parameter was a single column.

Reply:
Right, good catch. Thanks!